src/settings/OwnerSettings.test.js

```diff
@@ -1,35 +1,196 @@
+          holdShelfExpiryPeriod: { duration: 3, intervalId: 'Weeks' },
+          id: '3a40852d-49fd-4df2-a1f9-6e2641a6e91f',
+          locationIds: [],
+          metadata: { createdDate: '2021-10-14T03:22:53.912+00:00', updatedDate: '2021-10-14T03:22:53.912+00:00' },
+          name: 'Circ Desk 1',
+          pickupLocation: true,
+          staffSlips: [],
+        }]
+      }
+    ],
+    resource: 'ownerServicePoints',
+  },
+  owners: {
+    records: owner,
+    resource: 'owners',
+  }
+};
+
+const mutator = {
+  activeRecord: {
+    replace: jest.fn(),
+    update: jest.fn(),
+  },
+  refdataValues: {
+    DELETE: jest.fn(),
+    POST: jest.fn(),
+    PUT: jest.fn(),
+    cancel: jest.fn(),
+  },
+  updaterIds: {
+    replace: jest.fn(),
+    update: jest.fn(),
+  },
+  updaters:{
+    DELETE: jest.fn(),
+    POST: jest.fn(),
+    PUT: jest.fn(),
+    cancel: jest.fn(),
+  },
+  values:{
+    DELETE: jest.fn().mockReturnValue(Promise.resolve()),
+    POST: jest.fn(),
+    PUT: jest.fn(),
+    cancel: jest.fn(),
+  }
+
+};
+
+const propData = {
+  stripes: {
+    connect: (Component) => props => <Component {...props} resources={resources} mutator={mutator} />,
+    hasPerm: jest.fn().mockResolvedValue(true),
+  },
+  resources: localResources,
+  match: { path: '/settings/users/asdf' },
+  location: { pathname: '/settings/users/asdf' },
+  intl: {},
+  okapi: {
+    url: 'https://folio-testing-okapi.dev.folio.org',
+    tenant: 'diku',
+    okapiReady: true,
+    authFailure: [],
+    bindings: {},
+  },
+};
+
 describe('Owner settings', () => {
-  it('renders', () => {
-    const props = {
-      resources: {},
-      match: { path: '/settings/users/asdf' },
-      location: { pathname: '/settings/users/asdf' },
-      mutators: {},
-      okapi: {
-        url: 'https://folio-testing-okapi.dev.folio.org',
-        tenant: 'diku',
-        okapiReady: true,
-        authFailure: [],
-        bindings: {},
-      },
-    };
-    renderOwnerSettings(props);
-
-    expect(screen.getByTestId('controlled-vocab')).toBeTruthy();
+  beforeAll(() => {
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: jest.fn().mockImplementation(query => ({
+        matches: false,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(), // Deprecated
+        removeListener: jest.fn(), // Deprecated
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      }))
+    });
+  });
+  it('Component must be rendered', () => {
+    renderOwnerSettings(propData);
+    expect(screen.getByText('ui-users.owners.columns.desc')).toBeTruthy();
+    expect(screen.getByText('ui-users.owners.columns.asp')).toBeTruthy();
+  });
+  it('Delete functionality in component', () => {
+    act(() => {
+      renderOwnerSettings(propData);
+    });
+    act(() => {
+      fireEvent.click(document.querySelector('[id="clickable-delete-settings-owners-1"]'));
```

Comment:
@Mathansivavelmurugesan could you please replace `fireEvent` with `click` from `https://testing-library.com/docs/ecosystem-user-event` (that lib should already in place in ui-users). That should allow you to remove all these `act` wrappers.

Reply:
Also here is a good resource about the `act` and the warnings we sometimes see and how to approach it:

https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning